src/Build/BackEnd/Shared/TargetResult.cs

```diff
@@ -284,6 +280,68 @@ private void RetrieveItemsFromCache()
             }
         }
 
+        private void TranslateItems(ITranslator translator)
+        {
+            var itemsCount = _items?.Length ?? 0;
+            translator.Translate(ref itemsCount);
+
+            if (translator.Mode == TranslationDirection.WriteToStream)
+            {
+                // We will just calculate a very rough starting buffer size for the memory stream based on the number of items and a
+                // rough guess for an average number of bytes needed to store them compressed.  This doesn't have to be accurate, just
+                // big enough to avoid unnecessary buffer reallocations in most cases.
+                var defaultCompressedBufferCapacity = _items.Length * 64;
+                // Again, a rough calculation of buffer size, this time for an uncompressed buffer.  We assume compression 
+                // will give us 2:1, as it's all text.
```

Comment:
nit: It looks like the comment applied to the `DeflateStream` compression in the old code and is no longer relevant. Also, the variable is not used for anything else than calculating the uncompressed buffer capacity. Suggestion: simplify by eliminating this local and also remove the "compressed"/"uncompress" from identifier names.

Reply:
Fixed